crates/assists/src/handlers/replace_impl_trait_with_generic.rs

```diff
@@ -0,0 +1,161 @@
+use syntax::ast::{self, edit::AstNodeEdit, make, AstNode, GenericParamsOwner};
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+// Assist: replace_impl_trait_with_generic
+//
+// Replaces `impl Trait` function argument with the named generic.
+//
+// ```
+// fn foo<G>(bar: <|>impl Bar) {}
+// ```
+// ->
+// ```
+// fn foo<B: Bar>(bar: B) {}
+// ```
+pub(crate) fn replace_impl_trait_with_generic(
+    acc: &mut Assists,
+    ctx: &AssistContext,
+) -> Option<()> {
+    let type_impl_trait = ctx.find_node_at_offset::<ast::ImplTraitType>()?;
+    let type_param = type_impl_trait.syntax().parent().and_then(ast::Param::cast)?;
+    let type_fn = type_param.syntax().ancestors().find_map(ast::Fn::cast)?;
+
+    let impl_trait_ty = type_impl_trait
```

Comment:
I think this can be `type_impl_trait.type_bound_list()`, to handle cases like this one:

```
fn foo(x: impl Foo + Bar) {}
->
fn foo<I: Foo + Bar>()(x: I) {}
```